### PR TITLE
Simplify entity change interface

### PR DESCRIPTION
**Pull request by dimagorelik**
Remove redundant operation from the interface.
Fix API method names

### CODE REVIEW

File: main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java [hunk 1]
```diff
@@ -11,16 +10,36 @@
...
+     */
     SupportedChangeOperation getSupportedChangeOperation();
 
-    Stream<? extends EntityField<?, ?>> getFieldsToFetch(ChangeOperation changeOperation);
-
-    ValidationError validate(EntityChange<E> entityChange, Entity entity, ChangeOperation changeOperation);
+    /**
+     * @return a list of fields to fetch
+     */
+    Stream<? extends EntityField<?, ?>> fetchFields();
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java:32]: This name sounds like the method actually fetches fields.
I would rename to `fieldsToFetch()` or `fetchedFields()`

File: main/src/main/java/com/kenshoo/pl/entity/internal/validators/FieldsCombinationValidationAdapter.java [hunk 2]
```diff
@@ -59,7 +53,7 @@ public ValidationError validate(EntityChange<E> entityChange, Entity entity, Cha
     }
 
     private boolean hasSubstitutions() {
-        return validator.substitutions().findAny() != null;
+        return validator.substitutions().findAny().isPresent();
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/internal/validators/FieldsCombinationValidationAdapter.java:56]: :+1:

File: main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java (hunk 1 above)
Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java:32]: @effiban 
Generally, I agree with you, but it is the property name in all our Interfaces.

Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java:32]: @effiban 
Method name - fetchFields

Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/internal/validators/EntityChangeValidator.java:32]: OK I see it is really in many other places, unfortunately :(

**Pull request merged**